.github/workflows/test-integration-postgres-redis-rabbit-matrix.yml

```diff
@@ -0,0 +1,152 @@
+        default: 'npm run test:ci'
+        required: false
+        type: string
+      setup_global_postgres:
+        description: 'Setup Postgres @openphone/database repo'
+        default: true
+        required: false
+        type: boolean
+      global_postgres_ref:
+        description: 'Branch of Postgres @openphone/database repo to use'
+        default: 'master'
+        required: false
+        type: string
+      migration_command:
+        description: 'The command to run the Postgres migrations'
+        default: 'npm run db:up'
+        required: false
+        type: string
+      skip_migration:
+        description: 'Whether or not the migrations should be skipped. Useful for when you want to run specific migration tests'
+        default: false
+        required: false
+        type: boolean
+      postgres_version:
+        description: 'The version of Postgres to use'
+        default: '13'
+        required: false
+        type: string
+      postgres_user:
+        description: 'The Postgres user'
+        default: 'openphone'
+        required: false
+        type: string
+      postgres_password:
+        description: 'The Postgres password'
+        default: 'docker'
+        required: false
+        type: string
+      postgres_host:
+        description: 'The Postgres host'
+        default: 'localhost'
+        required: false
+        type: string
+      postgres_database:
+        description: 'The Postgres database'
+        default: 'openphone'
+        required: false
+        type: string
+      working-directory:
+        description: 'The path to the directory to run commands'
+        required: false
+        type: string
+        default: ./
+      timeout-minutes:
+        description: 'The max time to allow the action to run'
+        required: false
+        type: number
+        default: 10
+
+jobs:
+  test-integration-postgres-rabbit-redis-matrix:
+    runs-on: ubuntu-latest
+    timeout-minutes: ${{ inputs.timeout-minutes }}
+
+    services:
+      postgres-svc:
+        image: postgres:13
+        env:
+          POSTGRES_USER: ${{ inputs.postgres_user }}
+          POSTGRES_PASSWORD: ${{ inputs.postgres_password }}
+          POSTGRES_DB: ${{ inputs.postgres_database }}
+        ports:
+          - 5432:5432
+        options: >-
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+      redis-svc:
+        image: redislabs/rejson:latest
+        ports:
+          - 6379:6379
+        options: >-
+          --health-cmd "redis-cli ping"
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+      rabbitmq:
+        image: rabbitmq:3.9.7-management
+        env:
+          RABBITMQ_DEFAULT_USER: openphone
+          RABBITMQ_DEFAULT_PASS: docker
+        ports:
+          - 5672:5672
+          - 15672:15672
+    strategy:
+      matrix:
+        shard: [ 1, 2, 3, 4 ]
```

Comment:
Do we want to be able to configure the number of shards?

Reply:
I also just read that you can write your shards like this too: `shard: [1/3, 2/3, 3/3]` and that removes the need for `${{matrix.shard }}/${{ strategy.job-total }}`

No real preference on my end, however I wasn't able to find `job-total` on `strategy` in the GitHub docs: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstrategy

Reply:
Sure, can update it to that. `job-total` does work but I'm fine with it being how you suggested. 

https://github.com/OpenPhone/voice/actions/runs/6709319501/job/18232106341

Reply:
It would be really nice for the consumers of this workflow to only have to specify the number of shards, and not need to know exactly how GHA expects these to be formatted. It is possible to have
```
inputs:
  jest_test_shard_count:
    description: The number of Jest test shards to be run within a GitHub Action Matrix
    default: 0
    required: false
    type: number

strategy:
  matrix:
    shard: ${{ some-computation }}
```

Reply:
I'm already looking into how to do this, but yaml is quite limited in custom functionality so I'm not sure if it's possible.